### PR TITLE
docs: The Layout Process

### PR DESCRIPTION
**Pull request by Ombuweb**

### CODE REVIEW

**Review by ashutosh887 — APPROVED** [review]

File: content/guide/the-layout-process.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: The Layout process
+---
+
+Getting views to render with the right dimensions and positions requires a run of the layout process. When rendering, a recursive process runs through every view in the **view hierarchy** in two passes &mdash; a measure pass and a layout pass.
+
+During **the measure pass** every view is measured to obtain its desired size. The following properties are considered during the measuring:
+
+- `width`, `height`
+- `minWidth`, `minHeight`
```
Comment (SeanKelly369) [inline content/guide/the-layout-process.md:10]: Does minWidth and minHeight work in {N} now?

File: content/guide/the-layout-process.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- `visibility`
+- `marginTop`, `marginRight`, `marginBottom`, `marginLeft`
+
+During **the layout pass** every view is placed in a specific layout slot. The slot is determined by the result of the measure pass and the following properties:
+
+- `marginTop`, `marginRight`, `marginBottom`, `marginLeft`
+- `horizontalAlignment`, `verticalAlignment`
+
+The layout process is by nature an resource-intensive process and it's performance highly depends on the number views (and complexity).
+<!-- TODO: fix links -->
```
Comment (SeanKelly369) [inline content/guide/the-layout-process.md:20]: Reminder of the TODO

File: content/guide/the-layout-process.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+:::tip
+Try to keep the view hierarchy as flat as possible by utilizing different [Layout Containers](/ui-and-styling#layout-containers) rather than relying on excessive view nesting.
```
Comment (SeanKelly369) [inline content/guide/the-layout-process.md]:
```suggestion
It's a good practice to minimize deep nesting of views. Instead, utilize different [Layout Containers](https://chat.openai.com/ui-and-styling#layout-containers) to achieve a simpler and more organized view hierarchy. This approach improves readability, maintainability and performance.
```

File: content/guide/the-layout-process.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    <SomeItem />
+    <SomeItem />
+  </StackLayout>
+  <StackLayout orientation="horizontal">
+    <!-- ... -->
+  </StackLayout>
+</StackLayout>
+```
+
+<!--  -->
```
Comment (SeanKelly369) [inline content/guide/the-layout-process.md:41]: Comment was going to be added here?

File: content/guide/the-layout-process.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+```html
+<GridLayout rows="auto, auto" columns="auto, auto">
+  <SomeItem row="0" col="0" />
```
Comment (SeanKelly369) [inline content/guide/the-layout-process.md:45]: Is SomeItem are real tag name in {N}?

File: content/guide/the-layout-process.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  <SomeItem row="0" col="1" />
+  <!-- ... row="1" ... -->
+</GridLayout>
+```
+
+One-offs are fine with the `<StackLayout>` approach, but building a whole app with the first approach will usually result in poor performance.
```
Comment (SeanKelly369) [inline content/guide/the-layout-process.md]: Really?  I disagree.  StackLayout isn't terribly expensive, at least not compared to using GridLayout.

File: content/guide/the-layout-process.md (hunk 1 above)
Comment (rigor789) [reply content/guide/the-layout-process.md:10]: It should yeah: https://github.com/NativeScript/NativeScript/blob/7a353eca1481fc8ca3c23f36250aa2300011c89d/packages/core/ui/styling/style-properties.ts#L187-L221

File: content/guide/the-layout-process.md (hunk 4 above)
Comment (rigor789) [reply content/guide/the-layout-process.md:41]: This is used to not combine 2 code blocks into tabs, but instead keep them separate (at least in the old docs, subject to change in new setup - but should be fine to leave there).

File: content/guide/the-layout-process.md (hunk 6 above)
Comment (rigor789) [reply content/guide/the-layout-process.md]: Deeply nested StackLayouts would be more expensive than a single flat GridLayout

**Pull request merged**